addons/mail/static/src/core/common/notification_permission_service.js

```diff
@@ -1,10 +1,32 @@
 import { reactive } from "@odoo/owl";
 
 import { browser } from "@web/core/browser/browser";
-import { isAndroidApp, isIosApp } from "@web/core/browser/feature_detection";
+import {
+    isAndroidApp,
+    isDisplayStandalone,
+    isIOS,
+    isIosApp,
+} from "@web/core/browser/feature_detection";
 import { _t } from "@web/core/l10n/translation";
 import { registry } from "@web/core/registry";
 
+async function getIosPwaPermission() {
+    // requesting Notification.permission does not work on iOS: it always states "default"!
+    // iOS relies on serviceWorker.registration.pushManager which only works on HTTPS.
+    const registration = await browser.navigator.serviceWorker?.getRegistration();
+    if (!registration) {
+        if (browser.location.prototol !== "https:") {
+            return "denied";
+        }
+        return "prompt";
+    }
+    const subscription = await registration.pushManager.getSubscription();
+    if (subscription) {
+        return "granted";
+    }
```

Comment:
Isn't that redundant with what comes next? It should resolve to `prompt|denied|granted`